compiler/noirc_frontend/src/tests/metaprogramming.rs

```diff
@@ -74,3 +77,22 @@ fn unquoted_integer_as_integer_token() {
 
     assert_no_errors(src);
 }
+
+#[test]
+fn allows_references_to_structs_generated_by_macros() {
+    let src = r#"
+    comptime fn make_new_struct(_s: StructDefinition) -> Quoted {
+        quote { struct Bar {} }
+    }
+
+    #[make_new_struct]
+    struct Foo {}
+
+    fn main() {
+        let _ = Foo {};
+        let _ = Bar {};
+    }
```

Comment:
Was this the requested feature from the other team? Or by references did they mean in a function signature which we'd currently error for?